ts/features/idpay/onboarding/xstate/actions.ts

```diff
@@ -48,6 +53,23 @@ const createActionsImplementation = (
     navigation.navigate(IDPayOnboardingRoutes.IDPAY_ONBOARDING_FAILURE);
   };
 
+  const navigateToInitiativeMonitoringScreen = (context: Context) => {
+    if (context.initiative === undefined) {
+      throw new Error("initiative is undefined");
+    }
+
+    const parentNav:
+      | IOStackNavigationProp<AppParamsList, keyof AppParamsList>
+      | undefined = navigation.getParent();
+
+    parentNav?.replace(IDPayDetailsRoutes.IDPAY_DETAILS_MAIN, {
```

Comment:
I'm not familiar with xstate, what if parentNav is undefined? For ex. UX/UI side?

Reply:
If `parentNav`, for any reason, is undefined we can expect that there is no navigation to the target screen. UX/UI side we have no effects. 
Since we are in a nested navigator, we are talking about a very critical situation almost impossible to happen due the architecture of our navigation, but to be honest we could actually strengthen this system by passing the root navigator as well and use that instead.